docs/components/zeebe/deployment-guide/interceptors.md

```diff
@@ -0,0 +1,172 @@
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A simple interceptor that logs each incoming call. The class must be public
+ * since we will load it via JAR into the gateway.
+ */
+public final class LoggingInterceptor implements ServerInterceptor {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger("LoggingInterceptor");
+
+  @Override
+  public <ReqT, RespT> Listener<ReqT> interceptCall(
+      final ServerCall<ReqT, RespT> call,
+      final Metadata headers,
+      final ServerCallHandler<ReqT, RespT> next) {
+    final var listener = next.startCall(call, headers);
+    return new SimpleForwardingServerCallListener<>(listener) {
+      @Override
+      public void onMessage(final ReqT message) {
+        LOGGER.trace("intercepted a call");
+        super.onMessage(message);
+      }
+    };
+  }
+}
+```
+
+## Packaging an interceptor
+Next, you need to package the interceptor class into a fat JAR. Such a JAR must
+contain all classes (i.e. including all classes your own classes depend upon at
+runtime).
+
+There are many ways to do this, but for simplicity we'll use `javac` and `jar`
+directly. Note, that means we have to define a java manifest file by hand, in
+order to place the libraries' classes on the classpath.
+
+```sh
+# to compile LoggingInterceptor.java, we'll need to provide the grpc and slf4j api libraries
+javac -classpath .:lib/grpc-api.jar:lib/slf4j-api.jar ./LoggingInterceptor.java
+
+# both runtime libraries and the manifest must be packaged together with the compiled classes
+jar cvfm LoggingInterceptor.jar ./MANIFEST.MF ./*.class ./lib
+
+# let's verify the contents of the JAR
+jar tf ./LoggingInterceptor.jar
+# META-INF/
+# META-INF/MANIFEST.MF
+# LoggingInterceptor.java
+# LoggingInterceptor$1.class
+# lib/
+# lib/grpc-api.jar
+# lib/grpc.jar
+# lib/slf4j-api.jar
+# lib/slf4j.jar
+```
+
+## Loading an interceptor into a gateway
+An interceptor can be loaded into your gateway as a fat JAR. For each
+interceptor, you need to provide your gateway with:
+- an interception order index
+- an identifier to identify this specific interceptor
+- where to find the JAR with the interceptor class
+- the fully qualified name of the interceptor class
+
+Let's continue with the LoggingInterceptor example. We can provide these
+[configurations](components/zeebe/deployment-guide/configuration/configuration.md)
+using a gateway config file, environment variables or a mix of both. We'll be
+using a config file here.
+
+The following gateway config file configures our LoggingInterceptor so it can be
+loaded into the gateway at start-up.
+
+```yaml
+zeebe:
+  gateway:
+    ...
+
+    # allows specifying multiple interceptors
+    interceptors:
+
+      # determines the interception order
+      0:
```

Comment:
:wrench: This isn't necessary though - you can also just do:

```yaml
interceptors:
  - id: logging-interceptor
    className: LoggingInterceptor
    jarPath: /tmp/LoggingInterceptor.jar
```

The `0` is only necessary if you want to configure it via environment variables in the environment variable name.

Reply:
Oh interesting! I'll make to change the description accordingly